python/resdata/resfile/rd_restart_file.py

```diff
@@ -104,10 +105,7 @@ def assert_headers(self):
             else:
                 intehead_kw = self["INTEHEAD"][0]
                 doubhead_kw = self["DOUBHEAD"][0]
-                if "LOGIHEAD" in self:
-                    logihead_kw = self["LOGIHEAD"][0]
-                else:
-                    logihead_kw = None
+                logihead_kw = self["LOGIHEAD"][0] if "LOGIHEAD" in self else None
```

Comment:
This `else` block is never called from the tests. However, creating non-unified restart files is a bit involved. I can have a look at it.